### PR TITLE
Add Schema and Migration Files

### PR DESCRIPTION
**Pull request by lily4178993**
This pull request is raised to introduce the initial database schema as per the project requirements. The changes include the creation of the following tables:

1. **Users Table (`users`):**
   - `name` (string)
   - `photo_url` (string)
   - `bio` (text)
   - `postsCounter` (integer with default 0)
   - Timestamps (`created_at`, `updated_at`)
   - Photo link for the user's photo.

2. **Posts Table (`posts`):**
   - `title` (string)
   - `text` (text)
   - `commentsCounter` (integer with default 0)
   - `likesCounter` (integer with default 0)
   - `author_id` (foreign key referencing `users.id`)
   - Timestamps (`created_at`, `updated_at`)
   - Corresponding index on `author_id`.

3. **Comments Table (`comments`):**
   - `text` (text)
   - `user_id` (foreign key referencing `users.id`)
   - `post_id` (foreign key referencing `posts.id`)
   - Timestamps (`created_at`, `updated_at`)
   - Corresponding indexes on `user_id` and `post_id`.

4. **Likes Table (`likes`):**
   - `user_id` (foreign key referencing `users.id`)
   - `post_id` (foreign key referencing `posts.id`)
   - Timestamps (`created_at`, `updated_at`)
   - Corresponding indexes on `user_id` and `post_id`.

**Migration Files:**
- `db/migrate/20231205140630_create_users.rb`
- `db/migrate/20231205141210_create_posts.rb`
- `db/migrate/20231205145005_create_comments.rb`
- `db/migrate/20231205144647_create_likes.rb`

**Additional Notes:**
- Verified that table and column names align with the ERD diagram.
blog_app_erd_v1_1
- Included foreign keys in `posts`, `comments`, and `likes` tables as required.
- Added corresponding indexes for foreign key columns.

---

**Please review the changes, and feel free to provide feedback. Thank you!**

### CODE REVIEW

File: db/migrate/20231205140630_create_users.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class CreateUsers < ActiveRecord::Migration[7.1]
+  def change
+    create_table :users do |t|
+      t.string :name
+      t.string :photo
+      t.text :bio
+      t.integer :postsCounter, default: 0
```
Comment (zieeco) [inline db/migrate/20231205140630_create_users.rb:7]: - Hey, the naming convention in ruby uses the snake_case, kindly update the postsCounter to posts_counter.

File: db/migrate/20231205141210_create_posts.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class CreatePosts < ActiveRecord::Migration[7.1]
+  def change
+    create_table :posts do |t|
+      t.references :author, foreign_key: { to_table: :users, name: 'author_id' }
+      t.string :title
+      t.text :text
+      t.integer :commentsCounter, default: 0
+      t.integer :likesCounter, default: 0
```
Comment (zieeco) [inline db/migrate/20231205141210_create_posts.rb:8]: - Hey, the naming convention in ruby uses the snake_case, kindly update the commentCounter, likesCounter to likes_counter and comments_counter.

File: db/schema.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
+  add_foreign_key "comments", "posts"
+  add_foreign_key "comments", "users"
+  add_foreign_key "likes", "posts"
+  add_foreign_key "likes", "users"
+  add_foreign_key "posts", "users", column: "author_id", name: "author_id"
+end
```
Comment (zieeco) [inline db/schema.rb:61]: - After you've made the changes, kindly re-run your migration to effect the changes. `rails db:migrate`

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @lily4178993,





Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- Great effort creating the data models.✔️
- Good commit messages ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions


- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by lily4178993** [comment]
Thank you @zieeco for the thoughtful review and valuable feedback.

**Review by skyv26 — APPROVED** [review]
## Hi @lily4178993, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 



## Highlights
- Implemented the features as per the previous code reviewer suggestion ✅ 

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**